tests/Microsoft.Identity.Test.Unit/PublicApiTests/RefreshInTests.cs

```diff
@@ -124,12 +137,23 @@ public async Task ATS_NonExpired_NeedsRefresh_AADUnavailableResponse_Async()
                 harness.HttpManager.AddTokenResponse(TokenResponseType.Valid);
 
                 result = await app
-                  .AcquireTokenSilent(
-                      TestConstants.s_scope.ToArray(),
-                      account)
-                  .ExecuteAsync(CancellationToken.None)
-                  .ConfigureAwait(false);
+                    .AcquireTokenSilent(
+                        TestConstants.s_scope.ToArray(),
+                        account)
+                    .ExecuteAsync(CancellationToken.None)
+                    .ConfigureAwait(false);
                 Assert.IsNotNull(result);
+                
+                YieldTillSatisfied(() => harness.HttpManager.QueueSize == 0);
+                // Use reflection to get the value and wait till achieved
+                Type httpTeleMgrType = typeof(TokenCacheAccessRecorder);
```

Comment:
`TokenCacheAccessRecorder` is not part of the product. It's test. You can make anything you want from it public.